daml/Marketplace/Trading/Service.daml

```diff
@@ -123,24 +109,23 @@ template Service
               create order with status = Order.PendingExecution; providerOrderId = Some providerOrderId
             _ -> return orderCid
 
-      nonconsuming RejectOrderRequest : (ContractId Order.T, Order.TradeCollateral)
+      nonconsuming RejectOrderRequest : ContractId Order.T
         with
           createOrderRequestCid : ContractId CreateOrderRequest
           errorCode : Int
           errorMessage : Text
         do
-          CreateOrderRequest{details, collateral} <- fetch createOrderRequestCid
+          CreateOrderRequest{details} <- fetch createOrderRequestCid
           (orderCid, order) <- fetchByKey (provider, details.id)
           archive orderCid
 
+          unlockCollateral order.details.marketType
+          whenSome details.optExchangeFee \depositCid -> unlockDeposit depositCid
           let reason = Error with code = show errorCode; message = errorMessage
-          orderCid <- create order with status = Order.Rejected with reason
-          collateral <- withdrawFromAllocationAccount collateral
-          optExchangeFee <- withdrawFeesFromAllocationAccount details.optExchangeFee
 
-          return (orderCid, collateral)
+          create order with status = Order.Rejected with reason
 
-      nonconsuming MarketOrderCancelRequest : (ContractId Order.T, Order.TradeCollateral)
+      nonconsuming MarketOrderCancelRequest : ContractId Order.T
         with
```

Comment:
Not related to this PR, but: why do we need this choice, besides the `CancelOrder` one?

Reply:
This is specific to how Exberry handles Market orders - when a Market order is submitted as `Fill Or Kill` or `Immediate Or Cancelled`, when Exberry can neither fill the order nor provide a partial fill then it submits a `Order Cancel Message` on its Market Data API (see "Order Cancel Message" at https://docs.exberry.io/market-data#response).

The `MarketOrderCancelRequest` choice can only be called via the original `Trading.CreateOrderRequest` template from the Exberry Adapter (the `CreateOrderRequest` contract is created to trigger the Exberry Adapter and gets consumed based on the response from Exberry). 

The `CancelOrder` choice is called when the user requests to cancel an order which kicks off an entirely different flow to and from Exberry (for example, in Exberry user initiated cancellations are send and received on a different endpoint from the Market Orders which get "cancelled").